esmvalcore/_data_finder.py

```diff
@@ -247,18 +292,24 @@ def get_rootpath(rootpath, project):
 
 def _find_input_dirs(variable, rootpath, drs):
     """Return a the full paths to input directories."""
-    project = variable['project']
+    project = variable["project"]
 
     root = get_rootpath(rootpath, project)
-    path_template = _select_drs('input_dir', drs, project)
+    path_template = _select_drs("input_dir", drs, project)
 
     dirnames = []
     for dirname_template in _replace_tags(path_template, variable):
         for base_path in root:
             dirname = os.path.join(base_path, dirname_template)
-            dirname = _resolve_latestversion(dirname)
-            if dirname is None:
-                continue
+            if variable["frequency"] == "fx" and "*" in dirname:
+                dirname = _resolve_wildcards_and_version(dirname, base_path,
+                                                         project, drs)
+                var_from_dir = dir_to_var(dirname, base_path, project, drs)
+                for (key, val) in variable.items():
+                    if val == '*':
+                        variable[key] = var_from_dir.get(key, '*')
+            else:
+                dirname = _resolve_latestversion(dirname)
             matches = glob.glob(dirname)
```

Comment:
This line tries to glob a `None`, which breaks glob. Try to skip it, if a None is returned
```suggestion
            if not dirname:
                continue
            matches = glob.glob(dirname)
```

Reply:
that's the one, hang on am gonna commit that from local repo, am running the tests now

Reply:
done